### PR TITLE
Devbranch

### PR DESCRIPTION
**Pull request by Shifna12Zarnaz**
Added PostCredentials in Vault.go and PostStoreCred in handler_storage.go

### CODE REVIEW

File: server/pkg/client/vault.go [hunk 1]
```diff
@@ -84,3 +85,20 @@ func (v *Vault) GetCert(secretName, customerID string) (map[string]string, error
 
 	return certMap, nil
 }
+func (v *Vault) PostCredentials(secretName, username, password, astradbcreds string) error {
```
Comment (srikrishnabh1) [inline server/pkg/client/vault.go:88]: you should also write get credential func else how can we get data?

File: server/pkg/handler/handle_storage.go [hunk 2]
```diff
@@ -4,19 +4,21 @@ import (
...
 
-func (a *APIHandler) PostStoreCred(c *gin.Context) {
+func (a *APIHandler) PostStoreAgentCred(c *gin.Context) {
 	a.log.Debugf("Install Deploy applicaiton api invocation started")
 	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Minute)
 	defer cancel()
 
-	var req api.StoreCredRequest
+	//var req api.StoreCredRequest
+	var req api.StoreAgentCredRequest
```
Comment (srikrishnabh1) [inline server/pkg/handler/handle_storage.go:21]: I dont see vault has been called here.

**Review by srikrishnabh1 — CHANGES_REQUESTED** [review]

**Comment by Shifna12Zarnaz** [comment]
Added the get credentials and postcredentials in vault.go and respective handlers

File: server/pkg/handler/handle_storage.go [hunk 3]
```diff
@@ -57,3 +60,118 @@ func (a *APIHandler) PostStoreCred(c *gin.Context) {
...
+	_, cancel := context.WithTimeout(context.Background(), 5*time.Minute)
+	defer cancel()
+	var req api.StoreCredRequest
+	if err := c.BindJSON(&req); err != nil {
+		a.setFailedResponse(&c, "Failed to parse payload", err)
+		return
+	}
+
+	secretName := *req.Credname
+	astradbcreds := "secretname/astrabdcreds"
```
Comment (srikrishnabh1) [inline server/pkg/handler/handle_storage.go:114]: dont hardcode this

File: server/pkg/handler/handle_storage.go [hunk 4]
```diff
@@ -57,3 +60,118 @@ func (a *APIHandler) PostStoreCred(c *gin.Context) {
...
+	// 	Data: &api.CredentialsData{
+	// 		Username: &username,
+	// 		Password: &password,
+	// 	},
+	// })
+	a.log.Infof("Username:%v,Password:%v", username, password)
+	a.log.Debugf("Credentials retrieved successfully from Vault")
+}
+
+// func (a *APIHandler) PostStoreCred(c *gin.Context) {
```
Comment (srikrishnabh1) [inline server/pkg/handler/handle_storage.go:140]: Why is this commented, we need this to store the credentials in the agent

**Review by srikrishnabh1 — COMMENTED** [review]
fix the review comments

**Pull request closed without merging**